chainer/optimizer.py

```diff
@@ -280,6 +280,8 @@ def update_core_chainerx(self, param):
 
         """
         grad_array = param.grad
+        if grad_array is None:
+            return
```

Comment:
I think it's better to write the logic to skip the update at `Parameter.update()`, instead of here.
In that case, maybe you can write `assert param.array is not None` at the beginning of this method.

I also think this method should not early-return even if `grad_array is None`.
Mathematically that is equivalent to `grad == 0`. That might mean something for some optimizers.
`update_core` should be called even in that case.